### DISCUSSION TITLE
firebase dynamic links

### DISCUSSION THREAD

**Discussion by mohamed000123** [invertase/react-native-firebase#6670] [Q&A]
i am using these packages in my app with this version
 "@react-native-firebase/analytics": "^12.1.0",
    "@react-native-firebase/app": "^12.1.0",
    "@react-native-firebase/crashlytics": "^12.1.0",
    "@react-native-firebase/messaging": "^12.1.0",

i want to add     "@react-native-firebase/dynamic-links":     in my app but i got dependency issues 
when i used it with the same version ("^12.1.0") it didn't work 
 when updating all packages i got this error while installing pods  >>> 
None of your spec sources contain a spec satisfying the dependency: `FirebaseCoreExtension (= 8.1.1)`.

**Comment by mikehardy** [comment]
Please update to current versions before requesting help, attempting to diagnose something on old versions is inefficient and not worth the time

Once you are on all current versions this should be fixed.

**Reply by mohamed000123** [reply]
the problem is when i tried to use the latest versions , i got this error while downloading pods 
>>> 
None of your spec sources contain a spec satisfying the dependency: FirebaseCoreExtension (= 8.1.1).

**Reply by mikehardy** [reply]
Something is wrong in your project. You likely have an override in your Podfile for the firebase-ios-sdk version or similar?
Either way you have a project-specific build failure and need to compare what a project should look like when initialized cleanly with native versions versus what your project actually looks like. This will generate a project that looks correct, from scratch: https://github.com/mikehardy/rnfbdemo/blob/main/make-demo.sh

**Reply by mohamed000123** [reply]
i still have this error when trying to upgrade to the latest version
is there a solution ? or should I do it natively  with swift  ? 

[!] CocoaPods could not find compatible versions for pod "FirebaseCoreExtension":
  In Podfile:
    RNFBMessaging (from `../node_modules/@react-native-firebase/messaging`) was resolved to 16.4.3, which depends on
      FirebaseCoreExtension (= 8.1.1)

None of your spec sources contain a spec satisfying the dependency: `FirebaseCoreExtension (= 8.1.1)`.

You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.

**Reply by mikehardy** [reply]
@mohamed000123 I believe my comment from 4 days ago is still accurate. There is no reason that version 16.4.3 of react-native-firebase would depend on FirebaseCoreExtension 8.1.1 when that module did not even exist then as far as I'm aware. It appears to have only come in to existence with firebase-ios-sdk v9 https://github.com/firebase/firebase-ios-sdk/commits/master/FirebaseCoreExtension.podspec

**Something is wrong with your project, likely the Podfile**. You'll need to fix it. react-native-firebase is not the problem here.
